### PR TITLE
Add WindowsProcessLauncherService

### PR DESCRIPTION
**Pull request by pierricgimmig**
WindowsProcessLauncherService takes care of launching a Windows
process, possibly spinning at entry point. If the process was indeed 
launched spinning, the service offers functionality to suspend it and 
then resume it.

### CODE REVIEW

File: src/ClientServices/WindowsProcessLauncherClient.cpp [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+  }
+
+  UpdateLaunchedProcessState(pid, LaunchedProcess::State::kExecutingOrExited);
+  return outcome::success();
+}
+
+void WindowsProcessLauncherClientImpl::UpdateLaunchedProcessState(
+    uint32_t pid, LaunchedProcess::State new_state) {
+  absl::MutexLock lock(&launched_processes_by_pid_mutex_);
+  LaunchedProcess& launched_process = launched_processes_by_pid_.at(pid);
```
Comment (danielfenner) [inline src/ClientServices/WindowsProcessLauncherClient.cpp:159]: The existence of pid in launched_processes_by_pid_ is guaranteed here, right? Maybe just add a ORBIT_CHECK.

**Review by danielfenner — APPROVED** [review]
lgtm, merely nitpicking...

File: src/ClientServices/WindowsProcessLauncherClient.cpp [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+
+  std::unique_ptr<orbit_grpc_protos::WindowsProcessLauncherService::Stub>
+      windows_process_launcher_service_;
+
+  absl::Mutex launched_processes_by_pid_mutex_;
+  absl::flat_hash_map<uint32_t, LaunchedProcess> launched_processes_by_pid_
+      ABSL_GUARDED_BY(launched_processes_by_pid_mutex_);
+};
+
+constexpr uint64_t kGrpcDefaultTimeoutMilliseconds = 3000;
```
Comment (dpallotti) [inline src/ClientServices/WindowsProcessLauncherClient.cpp:65]: The obvious question is: why three seconds? I would add a little comment on the arbitrariness of this.

File: src/ClientServices/WindowsProcessLauncherClient.cpp [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+  LaunchProcessResponse response;
+
+  *request.mutable_process_to_launch() = process_to_launch;
+
+  std::unique_ptr<grpc::ClientContext> context = CreateContext();
+  grpc::Status status =
+      windows_process_launcher_service_->LaunchProcess(context.get(), request, &response);
+
+  if (!status.ok()) {
+    ORBIT_ERROR("\"LaunchProcess\" grpc call failed: code=%d, message=%s", status.error_code(),
```
Comment (dpallotti) [inline src/ClientServices/WindowsProcessLauncherClient.cpp:92]: Just my OCD: grpc -> gRPC

File: src/ClientServices/WindowsProcessLauncherClient.cpp [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+    State state_ = State::kInvalid;
+    orbit_grpc_protos::ProcessInfo process_info_;
+  };
+
+  void UpdateLaunchedProcessState(uint32_t pid, LaunchedProcess::State new_state);
+
+  std::unique_ptr<orbit_grpc_protos::WindowsProcessLauncherService::Stub>
+      windows_process_launcher_service_;
+
+  absl::Mutex launched_processes_by_pid_mutex_;
```
Comment (dpallotti) [inline src/ClientServices/WindowsProcessLauncherClient.cpp:60]: So this class is supposed to be used from multiple threads? A little class comment explaining thread safety of this class would then be beneficial.

**Review by pierricgimmig — COMMENTED** [review]
@dpallotti , @danielfenner , many thanks for the review!

**Pull request merged**